src/optimization/utils/constraints.jl

```diff
@@ -71,7 +71,7 @@ _ẑᵢ₊₁(m, i) = affine_map(m[:network].layers[i], m[:z][i])
 
 # Any encoding passes through here first:
 function encode_network!(model::Model, network::Network, encoding::AbstractLinearProgram)
-    get!(object_dictionary(model), :network, network)
+    object_dictionary(model)[:network] =  network # stores model[:network]
```

Comment:
Strictly overwriting the network here could have consequences later (if the network is needed again).

Reply:
If you are calling encode_network! and passing a network, I think it's unlikely you are wanting to reference some old stored network -- it makes sense to me that you want to reference the network you just passed. And it causes errors when you try to encode a second network, yes. 

And yes the goal is to encode two networks simultaneously. Re the second change, JuMP won't overwrite variables, it will give them distinct names, so the code that was commented "should never happen" isn't needed and throws errors when trying to encode two networks into the same jump model.

Reply:
I don't disagree with anything here, just wondering if this leads to problems down the line when assuming something about `model[:network]`. I'll have a look.

Reply:
Did you take a look?